group/assignment5/AllUnknownAlphabets.java

```diff
@@ -0,0 +1,113 @@
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.HashSet;
+import java.util.LinkedList;
+
+public class AllUnknownAlphabets {
+
+    /**
+     * Takes a dictionary that contains ordered words and returns all the alphabets in a List of Character Lists
+     * @param dictionary list of string that language's words.
+     */
+    public static List<List<Character>> findAllUnknownAlphabets (List<String> dictionary) {
+        Map<Character, Set<Character>> dictionaryGraph = new HashMap<>();
+        Map<Character, Integer> ingoingEdgeMap = new HashMap<>();
+        Map<Character, Boolean> visited = new HashMap<>();
+
+        // Creating the char graph for the dictionary
+        for (int i = 0; i < dictionary.size(); i++) {
+            for (int j = 0; j < dictionary.get(i).length(); j++) {
+                if (!dictionaryGraph.containsKey(dictionary.get(i).charAt(j))) {
+                    char letter = dictionary.get(i).charAt(j);
+                    // adds all the chars in the dictionary into the maps
+                    dictionaryGraph.put(letter, new HashSet<>());
+                    // initially ingoingEdges to the chars are 0
+                    ingoingEdgeMap.put(letter, 0);
+                    // initially none of the nodes is visited
+                    visited.put(letter, false);
+                }
+            }
+        }
+
+        return findAllTopSorts(dictionary, ingoingEdgeMap, dictionaryGraph, visited);
+    }
+
+    /**
+     * Compares each pair of neighboring words to find the incoming edges to a character
+     * This is important for topological sorting and understanding the precedence of the characters
+     * @param dictionary list of string that language's words.
+     * @param ingoingEdgeMap takes the initialized 0 incoming edge maps for chars and after every comparison if
+     *        we find a precedence between letters increments the corresponding letters incoming edge
+     * @param dictionaryGraph takes the dictionary graph created to, it adds the chars that
+     *         have incoming edges to certain chars
+     * @param visited represents the evidence of the visited nodes and it is sent as parameter because of the time
+     *         complexity, to avoid going through all the chars again
+     * Returns all the alphabets to its caller function
+     */
+    private static List<List<Character>> findAllTopSorts(List<String> dictionary,
```

Comment:
I was first a bit confused, because I didn't realized that "Top" shall be short for "Topological". Perhaps it would be more clear, if you rename the method ;)